core/src/meta/structs.rs

```diff
@@ -0,0 +1,33 @@
+use std::collections::HashMap;
+
+use super::{methods::Method, types::Type};
+
+#[derive(Clone, Debug)]
+pub struct Struct {
+    pub name: String,
+    pub fields: HashMap<String, Type>,
```

Comment:
I think we only should extract fields for non opaque structs. Opaque structs should stay opaque, perhaps using a different type like `OpaqueName { dynamic: bool }`?

Reply:
Since we can't detect whether a type reference is opaque or not within a module, I added an `opaque` field to structs and introduced an API to dereference a `Type` by pulling structs from the global set. Also opened #4 for checks that we'll need to add to the tool later.

Reply:
Wait, I'm confused, type references will always just use `Named`, yes? they don't need to know if something is opaque or not aside from dyn stuff

Reply:
Should be resolved now!